app/models/middleware_server.rb

```diff
@@ -33,6 +37,21 @@ def evaluate_alert(alert_id, event)
   end
 
   def in_domain?
-    !middleware_server_group.nil?
+    middleware_server_group
+  end
+
+  def hawkular?
+    product == 'Hawkular'
+  end
+
+  def immutable?
+    hawkular? ||
+      ['Immutable', 'Is Immutable'].any? do |key|
```

Comment:
Wildfly/EAP agents present this key, while Hawkular servers use the 'Immutable' key.